src/databricks/labs/ucx/azure/locations.py

```diff
@@ -171,6 +174,7 @@ def run(self):
                 migrated_loc_urls.append(migrated_loc_url)
 
         leftover_loc_urls = [url for url in missing_loc_urls if url not in migrated_loc_urls]
+        self._principal_acl.apply_location_acl()
```

Comment:
on Azure, you're applying location acl during external location migration, but there is no circular dependencies, this is because Azure has a separate class for ext location migrations

Reply:
https://github.com/databrickslabs/ucx/issues/1572 created issue to track this